apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -35,7 +36,33 @@ public sealed class SpannerConnectionStringBuilder : DbConnectionStringBuilder
         public string DataSource
         {
             get => GetValueOrDefault("Data Source");
-            private set => this["Data Source"] = value;
+            private set => this["Data Source"] = ValidatedDataSource(value);
+        }
+
+        private string ValidatedDataSource(string dataSource)
```

Comment:
This should basically be a DatabaseName (as in the resource), right? If so, we already have parsing code in there.

There's a difference though - the "projects", "instances" and "databases" part would be case-sensitive. Do we definitely want to make that non-case-sensitive? It's tricky given that we want to be like GCP in some ways and like ADO.NET in others... I strongly suspect that the actual *ID* parts are going to be treated case-sensitively anyway.